Source/Core/DiscIO/VolumeDisc.cpp

```diff
@@ -20,6 +20,27 @@ std::string VolumeDisc::GetGameID(const Partition& partition) const
 {
   char id[6];
 
+  // Triforce games have their Game ID stored in the boot.id file
+  const FileSystem* file_system = GetFileSystem(partition);
+  if (file_system)
+  {
+    std::unique_ptr<FileInfo> file_info = file_system->FindFileInfo("boot.id");
+    if (file_info && !file_info->IsDirectory())
+    {
+      u8* bootid_buffer = new u8[file_info->GetTotalSize()];
+      if (Read(file_info->GetOffset(), file_info->GetTotalSize(), bootid_buffer, partition))
```

Comment:
I don't think you need the temporary buffer here; can't you just use `Read(file_info->GetOffset() + 0x30, sizeof(id), id, partition)` ?

(I guess this is the replacement for #10084?)